controllers/tenant_controller.go

```diff
@@ -577,6 +636,47 @@ func (r *TenantReconciler) syncNamespaces(tenant *capsulev1beta1.Tenant) (err er
 	return
 }
 
+func (r *TenantReconciler) syncNetworkPolicy(tenant *capsulev1beta1.Tenant, namespace string, keys []string) error {
+	if err := r.pruningResources(namespace, keys, &networkingv1.NetworkPolicy{}); err != nil {
+		return err
+	}
+	for i, spec := range tenant.Spec.NetworkPolicies.Items {
+		t := &networkingv1.NetworkPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("capsule-%s-%d", tenant.Name, i),
+				Namespace: namespace,
+			},
+		}
+		// getting NetworkPolicy labels for the mutateFn
+		tl, err := capsulev1beta1.GetTypeLabel(&capsulev1beta1.Tenant{})
+		if err != nil {
+			return err
+		}
+		nl, err := capsulev1beta1.GetTypeLabel(&networkingv1.NetworkPolicy{})
+		if err != nil {
+			return err
+		}
```

Comment:
This could be moved out of the loop

Reply:
We cannot, we need the index for building the label selector.

Reply:
i'm talking about these lines
```
		tl, err := capsulev1beta1.GetTypeLabel(&capsulev1beta1.Tenant{})
		if err != nil {
			return err
		}
		nl, err := capsulev1beta1.GetTypeLabel(&networkingv1.NetworkPolicy{})
		if err != nil {
			return err
		}
```
in all other funcs they are out of loop, as label names are static